### ISSUE TITLE
Idea: Make it possible to manually add channels via right-clicking a channel link and maybe make an online-database where people can collaborate on the list

### ISSUE THREAD

**Issue by TomArrow** [miscavage/De-Mainstream-YouTube-Extension#4]
So for one, it would be nice to be able to just add channels manually directly from the browser. For example, right-click channel link -> Block this one

Second, it would be nice if you could have two lists, "Personal blocks" and "General list". If one adds something to the "General list" it will be uploaded to a public database where it can be quickly approved by a team of moderators or voted on by other users and if it reaches a certain treshold (for example >90% upvotes), it gets added to an official list which you can opt-in to, which then your plugin can download and use as well.

These lists wouldn't even need all the logos and fancy stuff, it would be fine just having a plaintext list, because functionality > design.

**Comment by TomArrow** [comment]
Or even make it possible to have different community-edited lists. For example here's some things that would be nice too:
- Big corporations list
- Mainstream bands list (music videos and such)

**Comment by MewingLullaby** [comment]
Agreed there is already a discrepency on what MSM is and what should be blocked by default, I'd advise that we block MSM (corporations) by default but allow another list for individuals that users can manage for themselves

**Comment by llirium** [comment]
This is a great idea.  A personal blacklist would let us eliminate generally annoying channels that we wouldn't miss if they disappeared, outside of the scope of news media.

Also, the list is centred on American mainstream media.  I would list my national channels I'd like you to add, but this idea is better.  In the long run, it reduces the work for you more than this make-issue-and-add method you've chosen.

**Comment by alexandereliott** [comment]
> Or even make it possible to have different community-edited lists. For example here's some things that would be nice too:
> 
> * Big corporations list
> * Mainstream bands list (music videos and such)

Something like what ublock origin does? 
They have dozens of lists including ads, spam, malware domains, etc.
If we could get something like that here that'd be great

**Comment by daedalusEV** [comment]
https://i.imgur.com/2Jca5f3.png

Literally any corpo whose vid was artificially placed in the "2016 meltdown election" search results, they should be put in the built-in block list, or put in some configurable block list, or whatever.  This is the most glaring sign that YouTube has been tinkering with the results of _particular_ search queries.  We **know** the results for this search did not remotely look like this a year or two ago.

The image above is the search query by "relevance", which is the default option.  Switch it to query by "view count" and it should look more like this:

https://i.imgur.com/BKO6ycX.png

And even then, that Showtime video was not there last time I checked (1 month ago).  They continue to become more and more sneaky with how they insert these prioritized accounts.

**Comment by miscavage** [comment]
this is a neat idea. will have to add to the pipeline for a future version..
maybe i can have a list of channels somewhere demainstream.com or in the extension's options/settings page to select from to block.

**Comment by TomArrow** [comment]
Yep that would be a good start. I think like @alexandereliott said it should probably be thematic lists like in Ublock origin, because in the long run, if you put in all the mainstream channels, including all of the variations (official, then maybe UK and others for each country), you will end up with such a large number that it will be impractical or cumbersome to select individual channels by hand, I think.

**Comment by miscavage** [comment]
i have added the ability to right click and block a channel. ref commit: a66eb76 gif animation: https://github.com/miscavage/De-Mainstream-YouTube-Extension/blob/e845b99fe4bbfdf9278a617834fa54b61ed8e37f/lib/img/media/add.gif

**Comment by TomArrow** [comment]
Looks great, thanks!

**Comment by someuser-blip** [comment]
Hi miscavage,

The ability to right click and block a channel does not work for me. I do not have an entry in the contextual menu.

I installed the 1.0.4 version from Mozilla's website, and I use Firefox 73 on Linux, if that's helpful.

(thanks for that fantastic extension by the way)

**Comment by miscavage** [comment]
live now on chrome, firefox: https://twitter.com/MarkMiscavage/status/1229431420403306496

closing this issue. please open a new issue with new questions, concerns, problems

**Comment by TomArrow** [comment]
It seems I have got the correct update in Firefox (it shows the gif with the instructions to add channels), but I don't get the item in the context menu. Sorry, should I create a new issue for that or is it okay if I mention it here?

**Comment by miscavage** [comment]
@TomArrow yes i have noticed that issue. the quick fix would be for you to highlight the channel name before right-clicking... will work on a solution.
